packages/utils/telemetry-utils/src/telemetryTypes.ts

```diff
@@ -0,0 +1,137 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryBaseLogger, TelemetryEventCategory } from "@fluidframework/common-definitions";
+
+/**
+ * Property types that can be logged.
+ *
+ * @remarks Logging entire objects is considered extremely dangerous from a telemetry point of view because people can
+ * easily add fields to objects that shouldn't be logged and not realize it's going to be logged.
+ * General best practice is to explicitly log the fields you care about from objects.
+ */
+export type TelemetryEventPropertyTypeExt =
+    | string
+    | number
+    | boolean
+    | undefined
+    | (string | number | boolean)[];
+
+ /**
+  * A property to be logged to telemetry containing both the value and a tag. Tags are generic strings that can be used
+  * to mark pieces of information that should be organized or handled differently by loggers in various first or third
+  * party scenarios. For example, tags are used to mark PII that should not be stored in logs.
+  */
+export interface ITaggedTelemetryPropertyTypeExt {
+    value: TelemetryEventPropertyTypeExt;
+    tag: string;
+}
+
+/**
+ * JSON-serializable properties, which will be logged with telemetry.
+ */
+export interface ITelemetryPropertiesExt {
+    [index: string]: TelemetryEventPropertyTypeExt | ITaggedTelemetryPropertyTypeExt;
+}
+
+/**
+ * Base interface for logging telemetry statements.
+ * Can contain any number of properties that get serialized as json payload.
+ * @param category - category of the event, like "error", "performance", "generic", etc.
+ * @param eventName - name of the event.
+ */
+ export interface ITelemetryBaseEventExt extends ITelemetryPropertiesExt {
```

Comment:
Let's call this `ITelemetryEventExt`, I want to reserve the "Base" terminology for the old types used across public interface boundaries.

Reply:
Or maybe the "Ext" version is not even needed